contracts/src/token/erc721/extensions/consecutive.rs

```diff
@@ -0,0 +1,1054 @@
+    /// Batch burn is not supported.
+    ForbiddenBatchBurn(ERC721ForbiddenBatchBurn),
+}
+
+impl MethodError for erc721::Error {
+    fn encode(self) -> alloc::vec::Vec<u8> {
+        self.into()
+    }
+}
+
+impl MethodError for checkpoints::Error {
+    fn encode(self) -> alloc::vec::Vec<u8> {
+        self.into()
+    }
+}
+
+// TODO: add an option to override these constants
+
+/// Maximum size of a batch of consecutive tokens. This is designed to limit
+/// stress on off-chain indexing services that have to record one entry per
+/// token, and have protections against "unreasonably large" batches of tokens.
+pub const MAX_BATCH_SIZE: U96 = uint!(5000_U96);
+
+/// Used to offset the first token id in
+/// [`Erc721Consecutive::_next_consecutive_id`]
+pub const FIRST_CONSECUTIVE_ID: U96 = uint!(0_U96);
+
+/// Consecutive extension related implementation:
+impl Erc721Consecutive {
+    /// Override of [`Erc721::_owner_of_inner`] that checks the sequential
+    /// ownership structure for tokens that have been minted as part of a
+    /// batch, and not yet transferred.
+    ///
+    /// # Arguments
+    ///
+    /// * `&self` - Read access to the contract's state.
+    /// * `token_id` - Token id as a number.
+    pub fn _owner_of_inner(&self, token_id: U256) -> Address {
+        let owner = self.__owner_of_inner(token_id);
+        // If token is owned by the core, or beyond consecutive range, return
+        // base value
+        if owner != Address::ZERO
+            || token_id < U256::from(FIRST_CONSECUTIVE_ID)
+            || token_id > U256::from(U96::MAX)
+        {
+            return owner;
+        }
+
+        // Otherwise, check the token was not burned, and fetch ownership from
+        // the anchors.
+        if self._sequential_burn.get(token_id) {
+            Address::ZERO
+        } else {
+            // NOTE: Bounds already checked. No need for safe cast of token_id
+            self._sequential_ownership.lower_lookup(U96::from(token_id)).into()
+        }
+    }
+
+    /// Mint a batch of tokens of length `batch_size` for `to`. Returns the
+    /// token id of the first token minted in the batch; if `batchSize` is
+    /// 0, returns the number of consecutive ids minted so far.
+    ///
+    /// Requirements:
+    ///
+    /// - `batchSize` must not be greater than [`MAX_BATCH_SIZE`].
+    /// - The function is called in the constructor of the contract (directly or
+    ///   indirectly).
+    ///
+    /// CAUTION: Does not emit a `Transfer` event. This is ERC-721 compliant as
+    /// long as it is done inside of the constructor, which is enforced by
+    /// this function.
+    ///
+    /// CAUTION: Does not invoke `onERC721Received` on the receiver.
+    ///
+    /// # Arguments
+    ///
+    /// * `&self` - Write access to the contract's state.
+    /// * `token_id` - Token id as a number.
+    ///
+    /// # Errors
+    ///
+    /// If to is [`Address::ZERO`] error [`rc721::Error::InvalidReceiver`] is
```

Comment:
Nit: IMHO we should follow the formatting and structure for code docs, like here https://github.com/OpenZeppelin/rust-contracts-stylus/blob/main/contracts/src/token/erc721/mod.rs#L277-L289